presto-spi/src/main/java/io/prestosql/spi/type/TypeUtils.java

```diff
@@ -51,6 +56,9 @@ public static Object readNativeValue(Type type, Block block, int position)
         if (javaType == Slice.class) {
             return type.getSlice(block, position);
         }
+        if (throwOnOtherPrimitives && javaType.isPrimitive()) {
```

Comment:
I'm not clear on why we need this

Reply:
I'll remove this refactor. It was inspired by the method visitDereferenceExpression() in ExpressionInterpreter which copied code of TypeUtils.readNativeValue() but added this check.